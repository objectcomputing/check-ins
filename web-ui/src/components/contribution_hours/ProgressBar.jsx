import React from "react";
import PropTypes from "prop-types";
import { makeStyles } from "@material-ui/core/styles";
import LinearProgress from "@material-ui/core/LinearProgress";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";

function LinearProgressWithLabel(props) {
  return (
    <Box display="flex" alignItems="center">
      <Box width="100%" mr={1}>
        <LinearProgress variant="buffer" {...props} />
      </Box>
    </Box>
  );
}

const propTypes = {
  /**
   * The value of the progress indicator for the determinate and buffer variants.
   * Value between 0 and 100.
   */
  billableHours: PropTypes.number,
  contributionHours: PropTypes.number.isRequired,
  targetHours: PropTypes.number.isRequired,
  ptoHours: PropTypes.number,
};

const useStyles = makeStyles({
  root: {
    width: "100%",
  },
});

const LinearBuffer = ({
  billableHours,
  contributionHours = 925,
  targetHours = 1850,
  ptoHours = 0,
}) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <LinearProgressWithLabel
        variant={billableHours ? "buffer" : "determinate"}
        value={
          billableHours
            ? (billableHours / targetHours) * 100
            : (contributionHours / targetHours) * 100
        }
        valueBuffer={
          billableHours ? (contributionHours / targetHours) * 100 : null
        }
      />
      <Typography
        align="right"
        variant="body2"
        color="textSecondary"
        style={{ display: "block" }}
      >
        {billableHours && <span>Billable Hours: {billableHours} - </span>}
        Contribution Hours: {contributionHours} - Target Hours: {targetHours}
      </Typography>
      <Typography
        align="right"
        variant="body2"
        color="textSecondary"
        style={ptoHours ? { display: "block" } : { display: "none" }}
      >
        PTO Hours: {ptoHours}
      </Typography>
    </div>
  );
};
LinearBuffer.propTypes = propTypes;
export default LinearBuffer;
