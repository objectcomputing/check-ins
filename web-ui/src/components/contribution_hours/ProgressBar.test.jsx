import React from 'react';
import ProgressBar from './ProgressBar';
import renderer from 'react-test-renderer';

it('has billable hours', () => {
  snapshot(<ProgressBar billableHours={1200}
                        contributionHours={1500}
                        targetHours={1850} />);
});

it('has no billable hours', () => {
  snapshot(<ProgressBar billableHours={0}
                        contributionHours={1500}
                        targetHours={1850} />);
});
