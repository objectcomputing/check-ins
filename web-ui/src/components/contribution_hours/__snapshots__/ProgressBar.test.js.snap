// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has billable hours 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <div
      className="MuiBox-root MuiBox-root-3"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={50}
        className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style={
            Object {
              "transform": "translateX(-50%)",
            }
          }
        />
      </div>
    </div>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignRight"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Billable Hours: 
    Contribution Hours: 
    925
     - Target Hours: 
    1850
     - PTO Hours: 
    0
  </p>
</div>
`;

exports[`has no billable hours 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-4"
  >
    <div
      className="MuiBox-root MuiBox-root-5"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={50}
        className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style={
            Object {
              "transform": "translateX(-50%)",
            }
          }
        />
      </div>
    </div>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-alignRight"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Billable Hours: 
    Contribution Hours: 
    925
     - Target Hours: 
    1850
     - PTO Hours: 
    0
  </p>
</div>
`;
